recipes/libressl/all/conanfile.py

```diff
@@ -1,73 +1,108 @@
 from conans import ConanFile, CMake, tools
+import glob
 import os
 
 
 class LibreSSLConan(ConanFile):
     name = "libressl"
+    provides = "openssl"
     url = "https://github.com/conan-io/conan-center-index"
     homepage = "https://www.libressl.org/"
     license = ("OpenSSL", "BSD", "ISC")
     settings = "os", "compiler", "build_type", "arch"
     options = {"shared": [True, False], "fPIC": [True, False]}
     default_options = {"shared": False, "fPIC": True}
     exports_sources = "CMakeLists.txt"
+    generators = "cmake"
     topics = ("SSL", "TLS", "openssl")
     description = ("LibreSSL is a version of the TLS/crypto stack forked from OpenSSL in 2014, "
                    "with goals of modernizing the codebase, improving security, and applying "
                    "best practice development processes.")
-    generators = "cmake"
-    _source_subfolder = "source_subfolder"
 
+    _cmake = None
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
 
     def config_options(self):
-        del self.settings.compiler.libcxx
-        del self.settings.compiler.cppstd
         if self.settings.os == "Windows":
             del self.options.fPIC
 
+    def configure(self):
+        if self.options.shared:
+            del self.options.fPIC
+        del self.settings.compiler.libcxx
+        del self.settings.compiler.cppstd
+
     def source(self):
         tools.get(**self.conan_data["sources"][self.version])
         extracted_dir = self.name + "-" + self.version
         os.rename(extracted_dir, self._source_subfolder)
 
+    def _configure_cmake(self):
+        if self._cmake:
+            return self._cmake
+        self._cmake = CMake(self)
+        self._cmake.definitions["LIBRESSL_SKIP_INSTALL"] = False
+        self._cmake.definitions["LIBRESSL_APPS"] = False # Warning: if enabled, do not use cmake installation, to avoid installing files in OPENSSLDIR
+        self._cmake.definitions["LIBRESSL_TESTS"] = False
+        self._cmake.definitions["ENABLE_ASM"] = True
+        self._cmake.definitions["ENABLE_EXTRATESTS"] = False
+        self._cmake.definitions["ENABLE_NC"] = False
+        self._cmake.definitions["OPENSSLDIR"] = "C:/Windows/libressl/ssl" if self.settings.os == "Windows" else "/etc/ssl"
```

Comment:
How to know if that path is correct? Is it just a placeholder?

Reply:
yeah, let's avoid hard-coded directories like that

Reply:
It's the default value in upstream CMakeLists, but it's indirectly set. If not set, it will point to `CMAKE_INSTALL_PREFIX/etc/ssl` or something like this.

See this disussion in original PR: https://github.com/conan-io/conan-center-index/pull/197#discussion_r337406334

By doing this, I preserve previous behaviour of this recipe, while allowing to use cmake install.

Reply:
Can you use `self.deps_cpp_info["openssl"].rootpath)` instead?

Reply:
@theodelrieu what do you think? In the original PR you said you had issues if this value was not `/etc/ssl`

Reply:
`/etc/ssl` to my knowledge is the standard for Unix based OS to look for manually installed certificates (when running as root) I presume they did something similar to make windows work but I haven't worked with any application that relied on those paths since I got into security 🤔 

I think maintaining the behavior is a very sound direction 👍 

I have a project lined up that could use this, so if there's anything blocking, let me know how I can help!

Reply:
@SpaceIm I did have issues with it, mainly my program could not load certificates. I honestly don't remember much from it...

Reply:
Is it the only point blocking this PR to be merged? This default value is also the one in upstream CMakeLists.txt. Couldn't it be merged and if there is any issue, it could be fixed in another PR?

Reply:
💯 for that!

Reply:
@theodelrieu @SpaceIm @prince-chrismc 
What do you think about my comment at https://github.com/conan-io/conan-center-index/issues/3476#issuecomment-727298157?